libgraphics/include/pivot/graphics/DescriptorAllocator/DescriptorLayoutCache.hxx

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <unordered_map>
+#include <vector>
+
+#include <vulkan/vulkan.hpp>
+
+namespace pivot::graphics
+{
+/// Cache the descriptor sets layout to avoid duplication
+class DescriptorLayoutCache
+{
+private:
+    struct DescriptorLayoutInfo {
+        std::vector<vk::DescriptorSetLayoutBinding> bindings;
+        bool operator==(const DescriptorLayoutInfo &other) const;
+        size_t hash() const;
```

Comment:
What is the purpose of this function ? What sort of hash is it computing ? Why do you need this ?

Reply:
See implementation